Kernel/Devices/USB/USBTransfer.cpp

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2021, Jesse Buhagiar <jooster669@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <Kernel/Devices/USB/USBTransfer.h>
+#include <Kernel/VM/ContiguousVMObject.h>
+#include <Kernel/VM/MemoryManager.h>
+
+namespace Kernel::USB {
+
+NonnullRefPtr<USBTransfer> USBTransfer::create(const USBPipe& pipe, u16 len)
+{
+    return adopt_ref(*new USBTransfer(pipe, len));
+}
+
+USBTransfer::USBTransfer(const USBPipe& pipe, u16 len)
+    : m_pipe(pipe)
+    , m_transfer_data_size(len)
+{
+    // Initialize data buffer for transfer
+    // This will definitely need to be refactored in the future, I doubt this will scale well...
+    auto vmobj = ContiguousVMObject::create_with_size(PAGE_SIZE);
```

Comment:
Should probably be created in `USBTransfer::create` and passed in through the constructor instead, so failure can be observed.